source/controllers/src/impedance/VelocityImpedance.cpp

```diff
@@ -0,0 +1,54 @@
+#include "controllers/impedance/VelocityImpedance.hpp"
+#include "state_representation/robot/JointState.hpp"
+#include "state_representation/robot/JointPositions.hpp"
+#include "state_representation/robot/JointVelocities.hpp"
+#include "state_representation/space/cartesian/CartesianState.hpp"
+#include "state_representation/space/cartesian/CartesianPose.hpp"
+#include "state_representation/space/cartesian/CartesianTwist.hpp"
+
+using namespace state_representation;
+
+namespace controllers::impedance {
+template<class S>
+VelocityImpedance<S>::VelocityImpedance(const Eigen::MatrixXd& stiffness,
+                                        const Eigen::MatrixXd& damping) :
+    Impedance<S>(stiffness, damping, Eigen::MatrixXd::Zero(stiffness.rows(), stiffness.cols())) {}
+
+template VelocityImpedance<CartesianState>::VelocityImpedance(const Eigen::MatrixXd&,
+                                                              const Eigen::MatrixXd&);
+template VelocityImpedance<JointState>::VelocityImpedance(const Eigen::MatrixXd&,
+                                                          const Eigen::MatrixXd&);
+
+template<class S>
+S VelocityImpedance<S>::compute_command(const S&, const S&) {
+  throw exceptions::NotImplementedException(
+      "compute_command(desired_state, feedback_state) not implemented for this input class");
+}
+
+template<>
+CartesianState VelocityImpedance<CartesianState>::compute_command(const CartesianState& desired_state,
+                                                                  const CartesianState& feedback_state) {
+  using namespace std::chrono_literals;
+  // compute the displacement by multiplying the desired twist by the unit time period and add it to the current pose
+  CartesianPose desired_pose= static_cast<CartesianPose>(feedback_state) + 1s * static_cast<CartesianTwist>(desired_state);
```

Comment:
When doing PR #96 I realized that to converge this line should be `CartesianPose desired_pose =  1s * static_cast<CartesianTwist>(desired_state) + static_cast<CartesianPose>(feedback_state);`. It is related to issue #27

Reply:
The main difference of this to the CITRIFIED twist controller is that the current pose is used here.

If we do the following, and explicitly set the feedback_state pose to Identity, then as far as I can tell the behaviour will be the same as the controller I have been using.
```suggestion
  CartesianPose desired_pose = 1s * static_cast<CartesianTwist>(desired_state);
```
For position, this "relative offset" doesn't matter anyway. For orientation, I already forgot the exact justification but I think it has to do with the angular displacement being in the "local frame" (from a null orientation) such that the vector part of the quaternion corresponds to the angular velocity we want

Reply:
Why does this relative offset not matter? The `integrated_desired_state` and the `feedback_state` could then have positions far away from each other, leading to a high control input in the `compute_command` no?